File: app/src/main/java/threading/ex01.java
package threading;

public class ex01 {
    static class NumberTask implements Runnable {
        private final String label;

        NumberTask(String label) {
            this.label = label;
        }

        @Override
        public void run() {
            for (int i = 1; i <= 10; i++) {
                System.out.println(label + " -> " + i);
                try {
                    Thread.sleep(30);
                } catch (InterruptedException e) {
                    Thread.currentThread().interrupt();
                    return;
                }
            }
        }
    }

    public static void main(String[] args) throws InterruptedException {
        Thread t1 = new Thread(new NumberTask("T1"));
        Thread t2 = new Thread(new NumberTask("T2"));

        t1.start();
        t2.start();

        t1.join();
        t2.join();
    }
}

File: app-kotlin/src/main/kotlin/threading/ex01.kt
package threading

class ex01 {
    class NumberTask(private val label: String) : Runnable {
        override fun run() {
            for (i in 1..10) {
                println("$label -> $i")
                try {
                    Thread.sleep(30)
                } catch (_: InterruptedException) {
                    Thread.currentThread().interrupt()
                    return
                }
            }
        }
    }

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            val t1 = Thread(NumberTask("T1"))
            val t2 = Thread(NumberTask("T2"))

            t1.start()
            t2.start()

            t1.join()
            t2.join()
        }
    }
}
